GoogleForCreators/web-stories-wp
# Publish Modal: Fix issues with apostrophes in site name

## Context



As reported at #10809, site titles with apostrophes in them are not correctly displayed in the pre-publish modal.

This is because WordPress saves them to the database as HTML entities, but we need the raw values.

## Summary



This PR fixes the issues with single quotes in the site name by using `html_entity_decode`

## Relevant Technical Choices



This PR also includes some other related tweaks I made while looking into this.

1. Call the `web_stories_publisher_name` filter earlier, within `load_from_post()`, so it doesn't run in a getter method. `get_publisher_name()` isn't supposed to modify the publisher name and run this filter every time it's called.
2. Do not escape publisher name / site title within `get_publisher_name`. Any escaping should be done as late as possible.
3. Do not escape `$lang` when setting the `html[lang]` attribute. The `setAttribute()` call does the escaping for us.

## To-do



## User-facing changes



**Before**

![Screenshot 2022-03-07 at 15 52 53](https://user-images.githubusercontent.com/841956/157067570-1438350e-0454-4423-baaf-c2e8662c0968.png)

**After**

![Screenshot 2022-03-07 at 16 13 40](https://user-images.githubusercontent.com/841956/157067604-b57b871f-9a5b-4d2a-88e1-3cc5676084e5.png)


## Testing Instructions






This PR can be tested by following these steps:

1. In the WordPress settings, change site title to include single/double quotes, e.g. `S'mores`.
4. Create a new story
5. Click on "Publish" to open the pre-publish modal
6. Verify that the site title is displayed correctly


## Reviews

### Does this PR have a security-related impact?



No.

Any attributes printed on the page are automatically escaped through DOMDocument usage in PHP.

On the client-side, escaping is done automatically through the DOM/React.

### Does this PR change what data or activity we track or use?



No

### Does this PR have a legal-related impact?



No

## Checklist




---



Fixes #10809
